docs/developer/developer-handbook.md

```diff
@@ -0,0 +1,93 @@
+# EDC Developer Handbook
+
+<!-- TOC -->
+* [EDC Developer Handbook](#edc-developer-handbook)
+  * [Introduction](#introduction)
+  * [Building a distribution](#building-a-distribution)
+    * [Seed some data](#seed-some-data)
+    * [Perform a data transfer](#perform-a-data-transfer)
+  * [The control plane - core concepts](#the-control-plane---core-concepts)
+    * [API objects in detail](#api-objects-in-detail)
+    * [EDC state machines](#edc-state-machines)
+    * [The extension model](#the-extension-model)
+    * [EDC dependency injection](#edc-dependency-injection)
+    * [Policy scopes and evaluation](#policy-scopes-and-evaluation)
+  * [The data plane](#the-data-plane)
+    * [Data plane selectors](#data-plane-selectors)
+    * [Writing a DataSink and DataSource extension](#writing-a-datasink-and-datasource-extension)
+    * [The Control API](#the-control-api)
+  * [Advanced concepts](#advanced-concepts)
+    * [Events and callbacks](#events-and-callbacks)
+    * [The EDC JUnit framework](#the-edc-junit-framework)
+    * [Automatic documentation](#automatic-documentation)
+    * [Customize the build](#customize-the-build)
+<!-- TOC -->
+
+## Introduction
+
+--> states the intended audience, required pre-existing knowledge, required tools, etc.
+
+## Building a distribution
+--> link to one of the samples
+
```

Comment:
Personally, I would put this part at the end of the document, but for an overview it may make sense at this point as well.

Reply:
my thinking was, that it's a developer's handbook, and developers usually want to play around with something, even before reading a bunch of text.